cratedb_toolkit/io/mongodb/export.py

```diff
@@ -65,6 +67,9 @@ def extract_value(value, parent_type=None):
     """
     if isinstance(value, dict):
         if len(value) == 1:
+            if "$binary" in value and value["$binary"]["subType"] in ["03", "04"]:
+                decoded = UUID(bytes=base64.b64decode(value["$binary"]["base64"]))
+                return extract_value(decoded, parent_type)
             for k, v in value.items():
                 if k.startswith("$"):
                     return extract_value(v, k.lstrip("$"))
```

Comment:
Sure enough, decoding MongoDB Extended JSON should not be performed manually, but should [use `_json_convert` instead](https://github.com/crate/commons-codec/blob/v0.0.14/src/commons_codec/transform/mongodb.py#L33-L62), like the CDC subsystem is doing it already.

Reply:
Through `decode_bson`, the machinery now uses BSON's `_json_convert`.

https://github.com/crate/cratedb-toolkit/blob/bc9362be5eab9d183b4b094008de1deb7a8fd00b/cratedb_toolkit/io/mongodb/copy.py#L52-L59